pkgs/tools/networking/yggdrasil/default.nix

```diff
@@ -0,0 +1,36 @@
+{ pkgs, lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "yggdrasil";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "yggdrasil-network";
+    repo = "yggdrasil-go";
+    rev = "v${version}";
+    sha256 = "0cbj9hqrgn93jlybf3mfpffb68yyizxlvrsh1s5q21jv2lhhjcwj";
+  };
+
+  modSha256 = "1r942mqlq9lqxizxi33768jp4j706mwa2fliz0qbkh98r4ri2292";
+
+  subPackages = [ "." ];
+
+  buildPhase = ''
+    export PKGNAME="${pname}"
+    export PKGVER="${version}"
+    ./build
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    cp {yggdrasil,yggdrasilctl} "$out/bin"
+  '';
+
+  meta = with lib; {
+    description = "An experiment in scalable routing as an encrypted IPv6 overlay network";
+    homepage = "https://yggdrasil-network.github.io/";
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ gazally ];
+    platforms = platforms.linux;
```

Comment:
```suggestion
```

Reply:
I've just force pushed a commit which I think addresses all your issues with the package.  Please take a look.  I'm still working on the service module.